GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -768,6 +792,31 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
             return result;
         }
 
+        private FileSystemTaskResult TryRemoveModifiedPath(string virtualPath, bool isFolder)
+        {
+            if (!this.newlyCreatedFileAndFolderPaths.Contains(virtualPath))
+            {
+                return FileSystemTaskResult.Invalid;

```

Comment:
Just noticed this now, but I believe that `FileSystemTaskResult.Invalid` is never supposed to be used in code (i.e. it's an invalid value for `FileSystemTaskResult`).

Given the choice I think it would be cleaner to move this check out of `TryRemoveModifiedPath` and that way the return value of `TryRemoveModifiedPath` reflects only whether we succeeded\failed when trying to update ModifiedPaths.

Reply:
I see that William commented on the `Invalid` value too, and I agree, from reading this code it seems like this check maybe just doesn't belong in here, at least as currently written.

I would probably change the usage of this method to something like:
```
    if (this.IsTempFile(virtualPath))
    {
        result = this.TryRemoveModifiedPath(virtualPath, isFolder);
    }
    else
    {
        result = this.TryAddModifiedPath(virtualPath, isFolder);
    }
```